controllers/actions.github.com/ephemeralrunner_controller.go

```diff
@@ -246,6 +246,13 @@ func (r *EphemeralRunnerReconciler) cleanupResources(ctx context.Context, epheme
 	switch {
 	case err == nil:
 		if pod.ObjectMeta.DeletionTimestamp.IsZero() {
+			if pod.Status.Phase == corev1.PodPending && ephemeralRunner.Status.RunnerId > 0 {
```

Comment:
I think we would just go with a second finalizer on the ephemeralrunner to handle removing the runner from the service.